src/__mocks__/Events.ts

```diff
@@ -0,0 +1,57 @@
+import {Ability, BuffEvent, BuffStackEvent, CastEvent} from 'fflogs'
+
+export const mockCastEvent = (timestamp: number, ability: Ability): CastEvent => ({
+	type: 'cast',
+	timestamp,
+	ability,
+	sourceIsFriendly: true,
+	targetIsFriendly: false,
+})
+
+export const mockApplyBuffEvent = (timestamp: number, ability: Ability, targetID: number): BuffEvent => ({
+	type: 'applybuff',
+	timestamp,
+	ability,
+	sourceIsFriendly: true,
+	targetIsFriendly: true,
+	targetID,
+})
+
+export const mockApplyBuffStackEvent = (timestamp: number, ability: Ability, targetID: number, stack: number): BuffStackEvent => ({
```

Comment:
at these sort of arg counts, it's usually better to move towards an object param. In cases of providing defaults &c, I usually do something like 

```ts
const createFoo = (overrides: Partial<Foo>): Foo => ({
	// write out
	// default values
	// here
	...overrides,
})
```

Reply:
I'm not sure I agree that it's worth the overhead to do a params object here - none of the parameters are or should be optional, and I would expect most IDEs/Intellisense to handle hinting what the parameters are anyway.  I could maybe see making the mock<T>Event stuff more generic, but I kind of suck with breaking down generics in my mental model and I'm not 100% sure it's really worth the candle to hoist a big generic structure for functions designed to provide syntactic sugar for creating mock events.